### PR TITLE
Add Unittests for Sound Channel Calculations and SSCP

### PR DESCRIPTION
**Pull request by NoahGallant-MUN**
Adds Unittests for the point functions for each of the sound speed calculations.
This means that it's testing that the correct value is being selected from the profile, but is not testing the looping through each profile.

This also adds sscp which is the possibility of a secondary sound channel at that point, and is either 0 if their isn't a possibility, or 1 if their is a possibility.
(This also has tests)

### CODE REVIEW

**Comment by htmlboss** [comment]
When creating PRs please ensure the assignes, labels, and project fields are filled out. It helps with tracking things down the road.

File: tests/test_functions.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        # Testing 2 sound channels, 2 critical depths
+        sspeed_4 = np.array([1435.76, 1434.12, 1434.98, 1436.27, 1434.51, 1433.97, 1433.49, 1433.67, 1435.09, 1435.98, 1436.12, 1437.41, 1438.19])
+        sca_idx = 6
+        sld_idx = 3
+
+        test_result = find_cd_idx(sca_idx, sld_idx, sspeed_4)
+        self.assertEqual(test_result, 10)
+        
+
+    #def test_cd_interpolation(self):
```
Comment (htmlboss) [inline tests/test_functions.py]: I think this decorator would help here: `@unittest.skip("reason for skipping")`. https://stackoverflow.com/a/16138561/2231969 so then you won't need to un-comment every line individually.

File: tests/test_functions.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        #sspeed_3 = np.array([1437.32, 1436.70, 1435.45, 1433.97, 1436.16, 1435.27, 1433.81, 1433.11, 1433.79, 1434.98, 1436.29, 1437.16, 1438.12])
+        #sld_idx = 3
+        #cd_idx = 10
+
+        
+        #sspeed_4 = np.array([1435.76, 1434.12, 1434.98, 1436.27, 1434.51, 1433.97, 1433.49, 1433.67, 1435.09, 1435.98, 1436.12, 1437.41, 1438.19])
+        
+
+    #def test_soniclayerdepth(self):
```
Comment (htmlboss) [inline tests/test_functions.py]: `@unittest.skip("reason for skipping")`

File: tests/test_functions.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+
+    #def test_criticaldepth(self):
```
Comment (htmlboss) [inline tests/test_functions.py]: `@unittest.skip("reason for skipping")`

File: tests/test_functions.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+
+    #def test_depthexcess(self):
```
Comment (htmlboss) [inline tests/test_functions.py]: @unittest.skip("reason for skipping")`

File: data/calculated_parser/functions.py [hunk 5]
```diff
@@ -226,6 +227,131 @@ def linearInterp(x1, y1, x2, y2, x):
 
     return cd_depth
 
+
+def __is_min(p1, p2, p3):
```
Comment (htmlboss) [inline data/calculated_parser/functions.py]: Is this function being used?

File: tests/test_functions.py (hunk 1 above)
Comment (NoahGallant-MUN) [reply tests/test_functions.py]: Good to know, thanks.
Will also keep that in mind for future pr's

File: data/calculated_parser/functions.py (hunk 5 above)
Comment (NoahGallant-MUN) [reply data/calculated_parser/functions.py]: No, switched to using scipy extrema function

**Pull request merged**